src/node.cc

```diff
@@ -320,7 +320,8 @@ MaybeLocal<Value> StartExecution(Environment* env, StartExecutionCallback cb) {
   CHECK(!env->isolate_data()->is_building_snapshot());
 
 #ifndef DISABLE_SINGLE_EXECUTABLE_APPLICATION
-  if (sea::IsSingleExecutable()) {
+  // Snapshot in SEA is only loaded for the main thread.
+  if (sea::IsSingleExecutable() && env->is_main_thread()) {
```

Comment:
Can we add a check that `env->snapshot_deserialize_main().IsEmpty()` when it is not the main thread?

I think this relies on the trick that `isolate->GetDataFromSnapshotOnce` will only return the same data once and the second time it will return empty, as the main thread and worker threads are using the same snapshot data.

Reply:
hmm yeah, we still haven't figured out how it should really work for workers - we'll likely need some restructuring of the snapshot blob/API. It currently goes like:

- default context with only stuff from V8
- vm context with the interceptor template and some fixups to the globals
- built-in main context which runs through CreateEnvironment()
- built-in worker context which currently only runs through NewContext()

All these share one isolate context.

It's still possible to support custom worker context snapshot + populate more stuff in built-in worker context snapshot, but it will probably need some restructuring. For a fuller built-in worker context snapshot we'll need to run an additional CreateEnvironment() call on the same isolate but with different flags and do this https://github.com/nodejs/node/blob/a243225aa2c016bc02471c5dd3bae3cf906f6d14/src/env.h#L568 , though I am not 100% sure if it works properly at the moment...can take a look as I look into the worker + snapshot combos a bit. For now a check to ensure snapshot_deserialize_main is not set on worker threads is fine.